### PR TITLE
feat(daemon): support additional admin UIDs via PEBBLE_ADMINS

### PR DESCRIPTION
**Pull request by hpidcock**
This PR proposes a means for the Pebble daemon to authenticate additional local unix users as admins through a new environment variable, `$PEBBLE_ADMINS`. This environment variable is a comma `,` delimited unordered list of UIDs that represent admins in addition to the defaults (e.g. root and the user running the daemon).

The use case for this is in Juju, where we have a charm container (the pebble client) and the workload container (the pebble daemon), both running as different non-root users. The charm container needs to be able to connect to and control the pebble process in the neighbouring container.

It is expected that the following should not work:
```
$ sudo pebble run &
$ pebble exec echo hello
error: access denied (try with sudo)
```

But with this PR, this is expected that the following should work:
```
$ sudo PEBBLE_ADMINS=1000 pebble run &
$ pebble exec echo hello
hello
```

[JU091](https://docs.google.com/document/d/1FKWkUvV4OPSh8BFzjmg0Jt0rVRHBTtZxzubcL5ls6H0/edit)

### CODE REVIEW

File: internals/cli/cli_test.go
Comment (benhoyt) [inline internals/cli/cli_test.go:190]: I think we should probably use `strutil.CommaSeparatedList`, which we use elsewhere in Pebble, to parse this. That does `strings.TrimSpace` on the values, which I think is a bit more user-friendly too. (We'd still have to whip over the list and convert to ints.)

File: internals/cli/cli_test.go
Comment (benhoyt) [inline internals/cli/cli_test.go:192]: I think we should only do a partial match of the error string, not test the full thing. Perhaps:

```go
    c.Assert(err, ErrorMatches, `cannot parse value " 345".*`)
```

File: internals/cli/cmd_run.go
Comment (benhoyt) [inline internals/cli/cmd_run.go:300]: I think this error message ends up a bit messy, with the `strconv.ParseUint` implementation details in there. What about dropping the `err` here and doing:

```go
fmt.Errorf("cannot parse PEBBLE_ADMINS: %q is not a valid user ID", uidStr)
```

File: internals/daemon/daemon.go
Comment (benhoyt) [inline internals/daemon/daemon.go:875]: I know two 0's in this list doesn't hurt if it's running as root, but it's also a bit strange. How about:

```suggestion
	d.admins = []sys.UserID{0}
	if uid := sysGetuid(); uid != 0 {
		d.admins = append(d.admins, uid)
	}
```

File: internals/daemon/daemon_test.go
Comment (benhoyt) [inline internals/daemon/daemon_test.go:472]: What about instead of adding this `daemonOpt` scaffolding, we just always add `0xbadd00d` to the list when we initialize the daemon in tests? (Similar above.)

**Review by benhoyt — COMMENTED** [review]
Looks good and simple! A few minor comments.

File: internals/cli/cli_test.go
Comment (hpidcock) [reply internals/cli/cli_test.go:190]: Great idea.

**Review by benhoyt — APPROVED** [review]
This change looks good to me now! Let's just get Gustavo's approval on the spec before merging.

**Pull request closed without merging**
